util.go

```diff
@@ -31,6 +32,37 @@ func PutHTTPClient(client *http.Client) {
 	httpClientPool.Put(client)
 }
 
+// WriteFileInfo represents the information necessary to write to a file.
+type WriteFileInfo struct {
+	Path string      `json:"path"`
+	Data []byte      `json:"data"`
+	Perm os.FileMode `json:"perm"`
+}
+
+// Validate ensure that all the parameters to write to a file are valid and sets
+// default permissions if necessary.
+func (info *WriteFileInfo) Validate() error {
+	catcher := grip.NewBasicCatcher()
+	if info.Path == "" {
+		catcher.New("path to file must be specified")
+	}
+
+	if info.Perm == 0 {
+		info.Perm = 0666
+	}
+	return catcher.Resolve()
+}
+
+// DoWrite writes the data to the given path, creating the directory hierarchy as
+// needed.
+func (info WriteFileInfo) DoWrite() error {
```

Comment:
pointer receiver?

Reply:
I changed it but ultimately, this doesn't really need a pointer receiver since it will only copy a few pointers + builtin types.